src/cheapestTree.ts

```diff
@@ -1,16 +1,30 @@
 import { calculateTreeQuantity } from './calculateTreeQuantity'
 import { calculateTreePrices } from './calculateTreePrices'
 import { calculateTreeCraftFlags } from './calculateTreeCraftFlags'
-import { RecipeTree, RecipeTreeWithCraftFlags } from './types'
+import { RecipeTree, RecipeTreeWithCraftFlags, RecipeTreeWithPrices } from './types'
 import { NestedRecipe } from '@gw2efficiency/recipe-nesting'
 
 export function cheapestTree(
   amount: number,
   tree: NestedRecipe,
   itemPrices: Record<string, number>,
   availableItems: Record<string, number> = {},
-  forceBuyItems: Array<number> = []
+  forceBuyItems: Array<number> = [],
+  valueOwnItems: string = 'false'
 ): RecipeTreeWithCraftFlags {
+  if (valueOwnItems === 'true') {
+    const treeWithQuantityWithoutAvailableItems = calculateTreeQuantity(
+      amount,
+      tree as RecipeTree,
+      {}
+    )
+    const treeWithPriceWithoutAvailableItems = calculateTreePrices(
+      treeWithQuantityWithoutAvailableItems,
+      itemPrices
+    )
+    const cheaperToBuyItemIds = getCheaperToBuyItemIds(treeWithPriceWithoutAvailableItems)
+    disableCraftForItemIds(tree, cheaperToBuyItemIds)
```

Comment:
Does this not have to be below L33?

Reply:
Ah, that makes sense. Could you add a quick oneliner comment above that ("calculateTreeQuantity already checks for craft flags, so we can adjust it here")